usr/iscsi_net_util.c

```diff
@@ -409,6 +440,127 @@ int net_setup_netdev(char *netdev, char *local_ip, char *mask, char *gateway,
 	return ret;
 }
 
+/**
+ * net_setup_netdev_ipv6 - bring up NIC for IPv6
+ * @netdev: network device name
+ * @local: ip address for netdev
+ * @prefix: prefix length
+ * @gateway: gateway
+ * @vlan: vlan ID
+ * @remote_ip: target portal ip
+ * @needs_bringup: bool indicating if the netdev needs to be started
+ *
+ * Bring up required NIC and use routing
+ * to force iSCSI traffic through correct NIC.
+ */
+int net_setup_netdev_ipv6(char *netdev, char *local_ip, int prefix, char *gateway,
+			  char *vlan, char *remote_ip, int needs_bringup)
+{
+	struct sockaddr_in6 sk_ipaddr = { .sin6_family = AF_INET6 };
+	struct sockaddr_in6 sk_gateway = { .sin6_family = AF_INET6 };
+	struct sockaddr_in6 sk_tgt_ipaddr = { .sin6_family = AF_INET6 };
+	struct in6_ifreq ifr6;
+	struct in6_rtmsg rtmsg;
+	char *physdev = NULL;
+	int sock;
+	int ret;
+	int vlan_id;
+	int ifindex;
+
+	if (!strlen(netdev)) {
+		log_error("No netdev name in fw entry.\n");
+		return EINVAL;
+	}
+
+	vlan_id = atoi(vlan);
+	if (vlan_id) {
+		physdev = netdev;
+		netdev = find_vlan_dev(physdev, vlan_id);
+	}
+	if (vlan_id && !netdev) {
+		log_error("No matching VLAN found for fw entry.\n");
+		return EINVAL;
+	}
```

Comment:
Thank you for your feedback. The logic aligns with the net_setup_netdev_ipv4 function, but your concern is valid. I propose introducing two new variables: vlandev and targetdev. The targetdev variable will initially point to netdev and switch to vlandev in VLAN scenarios. Ultimately, we can check the existence of vlandev instead of netdev. If you find this approach suitable, I will update the logic in both the net_setup_netdev_ipv4 and net_setup_netdev_ipv6 functions. Also, would you be okay with me combining these changes into a single commit?

Reply:
I see you have done a separate commit to fix the variable overloading issue. Thank you.